compiler/docgen.nim

```diff
@@ -1054,12 +1054,13 @@ proc writeOutput*(d: PDoc, useWarning = false) =
   if optStdout in d.conf.globalOptions:
     writeRope(stdout, content)
   else:
-    template outfile: untyped = d.destFile
```

Comment:
But this style avoided a copy.

Reply:
IIUC it's guaranteed to be a move in this case; for eg see below:
```nim
type Foo = distinct string
type Bar = ref object
  foo: Foo

proc main(a: Bar)=
  let b = a.foo
  doAssert cast[int](a.foo.string[0].unsafeAddr) == cast[int](b.string[0].unsafeAddr)

let a = Bar(foo: "abc".Foo)
main(a)
```

Reply:
That's a bug, finally addressed in --gc:arc, the move is unsound.

Reply:
* but `--gc:arc` also makes the assert pass
* I've changed it back to `template outfile: untyped = d.destFile` to unblock this PR;  `byRef: outfile = d.destFile` would be what I'd use once https://github.com/nim-lang/Nim/pull/11824 is merged; and it's also shorter than `template outfile: untyped = d.destFile`